### PR TITLE
bpo-44486: Make sure that modules always have a dictionary.

### PR DESCRIPTION
**Pull request by markshannon**
https://bugs.python.org/issue44486

### CODE REVIEW

**Review by gvanrossum — COMMENTED** [review]
Are,you *sure* you don’t just want to check for NULL in ceval.c? That seems so much simpler… Also the initialization of extension modules might be upset by this?

**Comment by markshannon** [comment]
Extension modules will be calling one of the C-API functions. They all create a dictionary.

**Review by gvanrossum — APPROVED** [review]
Okay, looks good.

File: Lib/test/test_module.py [hunk 1]
```diff
@@ -22,8 +22,8 @@ def test_uninitialized(self):
         # An uninitialized module has no __dict__ or __name__,
         # and __doc__ is None
         foo = ModuleType.__new__(ModuleType)
-        self.assertTrue(foo.__dict__ is None)
-        self.assertRaises(TypeError, dir, foo)
+        self.assertTrue(isinstance(foo.__dict__, dict))
```
Comment (terryjreedy) [inline Lib/test/test_module.py:25]: For next time, unittest has assertIsInstance(inst, class).

**Pull request merged**